src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -128,4 +132,31 @@ class GitVersionPlugin implements Plugin<Project> {
         // look in parent directory
         return scanForRootGitDir(currentRoot.parentFile)
     }
+
+    private static String runGitCommand(File dir, String ...commands) {
+        List<String> cmdInput = new ArrayList<>()
+        cmdInput.add("git")
+        cmdInput.addAll(commands)
+        ProcessBuilder pb = new ProcessBuilder(cmdInput)
+        pb.directory(dir)
+        pb.redirectErrorStream(true)
+
+        Process process = pb.start()
+        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))
+
+        StringBuilder builder = new StringBuilder()
+        String line = null
+        while ((line = reader.readLine()) != null) {
+            builder.append(line)
+            builder.append(System.getProperty("line.separator"))
+        }
+
+        int exitCode = process.waitFor()
+        if (exitCode != 0) {
+            return ""
```

Comment:
Yup, that is a totally valid concern.

Although I don't have concrete data, I'm fairly confident that the `git --describe --tag` behavior should be stable -- I think it would be a major back-compat break on their end if it was not.

The `describe` command and `--tags` flag have been around for as long as I can find online. The `--first-parent` flag was introduced in Git 1.8.4, which was released 08/23/2013.

The CircleCI nodes have version 2.7.4 of `git`, while locally I have 2.6.3. I think it's a fairly safe assumption that people have a version of Git that supports this.

The other option here is to make `--first-parent` an opt-in behavior, and to only use native `git` if we want that behavior. However, my other contention is that other tooling that wants to produce similar version strings will most likely shell out to `git` to do so, so in a way I would rather tie the behavior of this output to the output of `git` rather than to the particulars of the implementation of JGit.